src/web/core2avx.rkt

```diff
@@ -0,0 +1,261 @@
+  (define prec (ctx-lookup-prop ctx ':precision))
+  (define indent (ctx-lookup-extra ctx 'indent))
+  (define-values (name-ctx name) (ctx-random-name ctx prec))
+  (printf "~a~a ~a = ~a;\n" indent (type->avx prec) name (constant->avx x ctx))
+  (values name name-ctx))
+
+(define (visit-constant/avx vtor x #:ctx ctx)
+  (define prec (ctx-lookup-prop ctx ':precision))
+  (define indent (ctx-lookup-extra ctx 'indent))
+  (define-values (name-ctx name) (ctx-random-name ctx prec))
+  (printf "~a~a ~a = ~a;\n" indent (type->avx prec) name (constant->avx x ctx))
+  (values name 
+          (if (set-member? '(TRUE FALSE) x)
+              (ctx-update-props name-ctx (list ':precision 'boolean))
+              name-ctx)))
+
+(define-expr-visitor imperative-visitor avx-visitor
+  [visit-if visit-if/avx]
+  [visit-op_ visit-op_/avx]
+  [visit-call visit-op_/avx]
+  [visit-number visit-number/avx]
+  [visit-constant visit-constant/avx])
+
+(define (binary80->string x)
+  (parameterize ([gfl-exponent 15] [gfl-bits 80])
+    (let ([s (gfl->string (gfl x))])
+      (if (string-contains? s ".") s (string-append s ".0")))))
+
+(define (operator->avx out operator arguments context)
+  (define precision (ctx-lookup-prop context ':precision))
+  (define type (type->avx precision))
+  (define suffix (precision->suffix precision))
+  (define assign "~a = ~a")
+  (match operator
+    [(or '== '!= '< '> '<= '>=)
+     (define immediate 
+       (match operator
+         ['== "0x10"]
+         ['!= "0x1C"]
+         ['< "0x1"]
+         ['<= "0x2"]
+         ['> "0xE"]
+         ['>= "0xD"]))
+     (define comparisons
+       (if (equal? operator '!=)
+           ;; If the operator is `!=`, we need to compare each element with every
+           ;; other. What we want is the Cartesian product but with unordered
+           ;; instead of ordered pairs; and `(combinations <list> 2)` will give us this.
+           (combinations arguments 2)
+           ;; For the other operators, comparing elements stepwise (first with second,
+           ;; second with third, ...) and taking the conjunction will do the right thing.
+           (map list (drop-right arguments 1) (rest arguments))))
+     (define arguments*
+       (map
+        (match-lambda [(list a b) (format "_mm256_cmp_~a(~a, ~a, ~a)" suffix a b immediate)])
+        comparisons))
+     (format assign out (operator-nary->binary "and" suffix arguments*))]
+    ['and (format assign out (operator-nary->binary "and" suffix arguments))]
+    ['or (format assign out (operator-nary->binary "or" suffix arguments))]
+    ['not
+     (format "~a = ((~a) _mm256_xor_~a((__m256i) ~a, _mm256_set1_~a(-1LL)))" out type suffix (first arguments) suffix)]
+    ['round
+     (format "_mm256_round_~a(~a, ~a)"
+             suffix
+             (first arguments)
+             (round-mode->avx* (ctx-lookup-prop context ':round)))]
+    ['fabs
+     (format "_mm256_andnot_~a(_mm256_set1_~a(-0.0), ~a)" suffix suffix (first arguments))]
```

Comment:
Terrifying. Glad it works

Reply:
Lol. -0.0 is 0b1000000...000, not of that is 0b01111...111, and with that just forces the sign bit to be 0 meaning positive.